src/reducers/shelf/filter.ts

```diff
@@ -19,7 +19,8 @@ export function filterReducer(shelfSpec: Readonly<ShelfUnitSpec> = DEFAULT_SHELF
       const {filter} = action.payload;
       let index = action.payload.index;
       if (contains(shelfSpec.filters, filter)) {
-        throw new Error('Cannot add more than one filter of the same field');
+        window.alert('Cannot add more than one filter of the same field');
```

Comment:
Hmm, the reducer shouldn't cause side effect to the UI (via `window`). 

Instead, I think throwing error here is totally fine.  

What we need to design is where do we catch these errors and correctly throw error.